app/src/main/java/com/infomaniak/mail/ui/MainViewModel.kt

```diff
@@ -435,6 +450,9 @@ class MainViewModel @Inject constructor(
                 started = ::startedDownload,
                 stopped = ::stoppedDownload,
             )
+        } else if (isSwipe) {
+            // We need to make the swiped Thread come back, so we reassign the LiveData with Realm values
+            reassignCurrentThreadsLive()
```

Comment:
No we can't because we need to get the right data back from realm, and it's the only way retrigger the realm query

Reply:
Doesn't the data stay the same? Why could the data have changed?

Reply:
The data in realm is the same, but the data of the adapter is false, and our logic is to react to realm for the data of the adapter, so we need to make the realm live data trigger so the adapter get the update with the good data

Reply:
Why could the adapter data have changed? You just swiped the item to delete it, right? If so nothing should have changed and if something did indeed change in realm (because of like a notification we juste received), then realm would've already notified the item by itself without reassigning anything

Reply:
Because we manually deleted an item from the adapter

Reply:
They've already talked to me about this, and it seems to me that it's because you only remove the adapter element, so it's no longer synchronized with realm.
So when the action couldn't be performed, you'd have to synchronize with realm again. 
In other words, listen to realm again.

But if there's another way of doing it, I'd have to compare the impact with how it was done on **iOs**.
For now, we can keep it that way.